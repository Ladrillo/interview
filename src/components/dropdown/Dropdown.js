import React, { Component } from 'react';


export default function Dropdown({ options, value, changeHandler, className, name }) {
    return (
        <select name={name} className={className} onChange={changeHandler} value={value}>
        {
            options.map(option => (
                <option value={option.id} key={option.id}>{option.name}</option>
            ))
        }
        </select>
    );
}
