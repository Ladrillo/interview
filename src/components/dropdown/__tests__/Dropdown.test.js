import Dropdown from '../Dropdown';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';


describe('Dropdown', () => {
    const options = [
        { id: 1, name: 'foo' },
        { id: 2, name: 'bar' },
    ];
    const changeHandler = () => console.log('changing value...');
    const value         = 2;
Â 
    it('renders correctly', () => {
        const rendered = renderer.create(
            <Dropdown
                options       = {options}
                changeHandler = {changeHandler}
                value         = {value}
            />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
