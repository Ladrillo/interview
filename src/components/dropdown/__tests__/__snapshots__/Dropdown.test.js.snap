exports[`Dropdown renders correctly 1`] = `
<select
  onChange={[Function]}
  value={2}>
  <option
    value={1}>
    foo
  </option>
  <option
    value={2}>
    bar
  </option>
</select>
`;
